python/examples/xegpu_matmul/mlir_utils.py

```diff
@@ -0,0 +1,33 @@
+from mlir import ir
+from mlir.dialects import transform
+from mlir.dialects.transform import structured
+import os
+
+
+def apply_registered_pass(*args, **kwargs):
+    return transform.apply_registered_pass(transform.AnyOpType.get(), *args, **kwargs)
+
+
+def match(*args, **kwargs):
+    return structured.MatchOp(transform.AnyOpType.get(), *args, **kwargs)
+
+
+def cse(op):
+    transform.ApplyCommonSubexpressionEliminationOp(op)
+
+
+def canonicalize(op):
+    with ir.InsertionPoint(transform.ApplyPatternsOp(op).patterns):
+        transform.ApplyCanonicalizationPatternsOp()
+
+
+def get_mlir_library_path():
```

Comment:
This look up doesn't seem too flexible.
It might be better to pass libs through CLI or use ctypes utils to find libs through OS env.